jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -51,7 +51,7 @@ class OutlierDetectionStep(Step):
         weight_type = option('ivm','exptime',default='ivm')
         pixfrac = float(default=1.0)
         kernel = string(default='square') # drizzle kernel
-        fillval = string(default='INDEF')
+        fillval = string(default='NAN')
```

Comment:
Is this directly related to the spectral outlier detection or related to a different issue?  (I'm fine including it either way, but just would like to know what prompted this specifically)

Reply:
It is related - with some proposed parameter combinations, leaving fillval = INDEF does bad things to spectra, since it may combine zeros with real data.  NaN values are always ignored.

But also, the default for resampling was changed in #8488, and the outlier detection default should be changed to match, for safer and more consistent behavior.